src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1796,6 +1796,35 @@ public static void TestJsonReaderUtf8SpecialString(string jsonString, string exp
             }
         }
 
+        [Fact]
+        public static void TestBOMWithSingleJsonValue()
```

Comment:
As an aside, it would be good to keep some of these tests and actually invert the behavior. That is, validate we end up throwing if the UTF-8 BOM is present at the beginning of the `ROS<byte>`/`ROSequence<byte>` for both isFinalBlock = true and false, regardless of whether there is no more data after the BOM or if we have some valid JSON (because that is the expected behavior).